wasm/src/programs/manager/deploy.rs

```diff
@@ -207,4 +213,138 @@ impl ProgramManager {
             .saturating_mul(1_000_000); // 1 microcredit = 1e-6 credits.
         Ok(namespace_cost)
     }
+
+    /// Deploy an Aleo program without synthesizing keys and generating certificates.
+    /// Intended for use with Leo Devnode.
+    ///
+    /// @param private_key The private key of the sender
+    /// @param program The source code of the program being deployed
+    /// @param imports A javascript object holding the source code of any imported programs in the
+    /// form \{"program_name1": "program_source_code", "program_name2": "program_source_code", ..\}.
+    /// Note that all imported programs must be deployed on chain before the main program in order
+    /// for the deployment to succeed
+    /// @param priority_fee_credits The optional priority fee to be paid for the transaction
+    /// @param fee_record The record to spend the fee from
+    /// @param url The url of the Aleo network node to send the transaction to
+    /// @param imports (optional) Provide a list of imports to use for the program deployment in the
+    /// form of a javascript object where the keys are a string of the program name and the values
+    /// are a string representing the program source code \{ "hello.aleo": "hello.aleo source code" \}
+    /// @param fee_proving_key (optional) Provide a proving key to use for the fee execution
+    /// @param fee_verifying_key (optional) Provide a verifying key to use for the fee execution
+    /// @returns {Transaction}
+    #[wasm_bindgen(js_name = buildDevnodeDeploymentTransaction)]
+    #[allow(clippy::too_many_arguments)]
+    pub async fn devnode_deploy(
+        private_key: &PrivateKey,
+        program: &str,
+        priority_fee_credits: f64,
+        fee_record: Option<RecordPlaintext>,
+        url: Option<String>,
+        imports: Option<Object>,
+        offline_query: Option<OfflineQuery>,
+    ) -> Result<Transaction, String> {
+        log("Creating deployment transaction");
+        let mut process_native = ProcessNative::load_web().map_err(|err| err.to_string())?;
+        let process = &mut process_native;
+
+        log("Checking program has a valid name");
+        let program = ProgramNative::from_str(program).map_err(|err| err.to_string())?;
+        let program_id = program.id();
+
+        log("Checking program imports are valid and add them to the process");
+        ProgramManager::resolve_imports(process, &program, imports)?;
+        let rng = &mut StdRng::from_entropy();
+
+        log("Creating deployment");
+        let node_url = url.as_deref().unwrap_or(DEFAULT_URL);
+
+        // Create deployment without synthesizing keys and generating certificates.
+        assert!(!program.functions().is_empty(), "Program must have at least one function");
```

Comment:
This should throw an error instead otherwise this `assert!` will throw an opaque crash upward.